src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/ForgotPassword.razor.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Text.RegularExpressions;
+
+namespace TodoTemplate.App.Pages;
+
+public partial class ForgotPassword
+{
+    public string? Email { get; set; }
+
+    public string? EmailErrorMessage { get; set; }
+
+    public bool IsSubmitButtonEnabled { get; set; }
+    public bool IsLoading { get; set; }
+
+    [Inject] public HttpClient HttpClient { get; set; } = default!;
+
+    public BitMessageBarType ForgotPasswordMessageType { get; set; }
+    public string? ForgotPasswordMessage { get; set; }
+
+    private void CheckSubmitButtonEnabled()
+    {
+        if (string.IsNullOrEmpty(Email))
+        {
+            IsSubmitButtonEnabled = false;
+            return;
+        }
+        IsSubmitButtonEnabled = true;
+    }
+
+    private bool ValidateForgotPassword()
+    {
+        EmailErrorMessage = string.IsNullOrEmpty(Email)
+            ? "Please enter your email."
+            : null;
+
+        if (string.IsNullOrEmpty(EmailErrorMessage) is false)
+        {
+            return false;
+        }
+
+        var isCorrectEmailFormat = new Regex(@"^([\w\.\-]+)@([\w\-]+)((\.(\w){2,3})+)$");
+
+        EmailErrorMessage = isCorrectEmailFormat.Match(Email!).Success is false
+            ? "The email address format is incorrect."
+            : null;
+
+        if (string.IsNullOrEmpty(EmailErrorMessage) is false)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private async Task Submit()
+    {
+        if (IsLoading)
+        {
+            return;
+        }
+
+        IsLoading = true;
+        IsSubmitButtonEnabled = false;
+        ForgotPasswordMessage = null;
+
+        if (ValidateForgotPassword() is false)
```

Comment:
Check ValidateForgotPassword before setting IsLoading to true, so you won't have to set it to false again

Reply:
I changed the code as you said. Keep in mind this issue already exist on other razor pages like sign-in. @ysmoradi

Reply:
plz submit the issue for other page(s)

Reply:
> @RezaKargar Look at this, I think you'd be interested.

@ysmoradi  I just created an issue (#1747) for it, check it out please